src/id3_variants_training/ID3_Class.py

```diff
@@ -122,6 +126,40 @@ def is_leaf_node(self, subset, split_path, split_index):
     def print_tree(self, file_name):
         DotExporter(self.root_node, nodenamefunc=ID3_Node.name_func).to_picture(file_name)
 
+    @staticmethod
+    def calc_other_split_variant_counts(w_var_counts, subset):
+        """
+        Given a subset that has been split, and counts on one side of the split,
+        calculate counts on the other side.
+
+        In the normal case, this is trivial; but if the counts include differentially
+        private noise, w_var_counts[ancestry1] could exceed subset[ancestry1].
+
+        This routine prevents that by capping the counts counts in the (mutable)
+        dictionary w_var_counts, then doing the subtraction.
+
+        Args:
+            w_var_counts (dict): A dictionary containing keys of ancestries and counts on one side of the split
+            subset (dict): A dictionary containing keys of ancestries and counts in the parent
+
+        Returns:
+            w_var_counts(dict): A possibly updated w_var_counts, after capping
+            wo_var_counts(dict): Dictionary on other side of the count
+        """
+        new_w_var_counts = {}
+        wo_var_counts = {}
+        for key, value in w_var_counts.items():
+            subset_value = subset.get(key, 0)
```

Comment:
We could avoid all these get(key, 0)s if we use a collections.defaultdict(int) - not for this PR but maybe later if we do a code cleanup